core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/messaginglayer/OutputManager.scala

```diff
@@ -106,9 +108,11 @@ class OutputManager(
   ): Unit = {
     val partitioner =
       partitioners.getOrElse(outputPort, throw new RuntimeException("output port not found"))
-    val bucketIndex = partitioner.getBucketIndex(tuple)
-    val destination = partitioner.allReceivers(bucketIndex)
-    networkOutputBuffers((outputPort, destination)).addTuple(tuple)
+    val it = partitioner.getBucketIndex(tuple)
+    for (bucketIndex <- it) {
+      val destination = partitioner.allReceivers(bucketIndex)
```

Comment:
Try to use plural forms names for collections and singular form for a single instance.